### PR TITLE
AppAuth 1.6.2 release

### LINKED ISSUE #23261: [BUG] Access token parsing failed
**Describe the bug**
When I try to get an access token for an external resource using Client certificate token provider by using AzureServiceTokenProvider.GetAccessTokenAsync(), I receive an error saying the access token couldn't be parsed.

**Expected behavior**
We should not try to parse the access token since it is not in the format expected. 

**Actual behavior (include Exception or Stack Trace)**
Microsoft.Azure.Services.AppAuthentication.AzureServiceTokenProviderException: Parameters: Connection String: RunAs=App;AppId=e165870b-8f4f-48ad-951f-2752be329850;TenantId=cdc5836a-15c5-4db6-b079-fcadd2505dc2;CertificateSubjectName=CN=sp.m365insidexxxxx.gbl;CertificateStoreLocation=CurrentUser, Resource: api://04cbdc38-6aaf-4668-bc55-b12e857433, Authority: https://login.microsoftonline.com/cdc5836a-15c5-4db6-b079-fcadd2505dc2. Exception Message: Tried 1 certificate(s). Access token could not be acquired.
Exception for cert #1 with thumbprint 6E199A1F059912D67xxxxx: **Access token is not in the expected format. Exception: There was an error deserializing the object of type Microsoft.Azure.Services.AppAuthentication.AccessToken. Encountered unexpected character 'w'.**
   at Microsoft.Azure.Services.AppAuthentication.AzureServiceTokenProvider.<GetAuthResultAsyncImpl>d__19.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Services.AppAuthentication.AzureServiceTokenProvider.<GetAccessTokenAsync>d__21.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()

**To Reproduce**
`var azureServiceTokenProvider = new AzureServiceTokenProvider($"RunAs=App;AppId={"xxxxx"};TenantId={"zzzz"};CertificateSubjectName=CN={"xxxx"};CertificateStoreLocation=CurrentUser");`
`string token = await azureServiceTokenProvider.GetAccessTokenAsync("https:\resourcexx")`

**I was able to fix the issue by removing line 197: PrincipalUsed.TenantId = AccessToken.Parse(accessToken).TenantId; from ClientCertificateAccessTokenProvider.cs**
So, the issue is that the returned access token is not in the format expected by the library but is still a valid token.

**Environment:**
 - Name and version of the Library package used: Microsoft.Azure.Services.AppAuthentication 1.6.1
 - Hosting platform or OS and .NET runtime version (`dotnet --info` output for .NET Core projects): .NET Framework 4.6.3

### PR DESCRIPTION
**Pull request by anaismiller**
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

**Comment by anaismiller** [comment]
This PR is ready to be merged

File: sdk/mgmtcommon/AppAuthentication/Azure.Services.AppAuthentication/Microsoft.Azure.Services.AppAuthentication.csproj [hunk 1]
```diff
@@ -1,8 +1,10 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
     <PropertyGroup>
         <PackageId>Microsoft.Azure.Services.AppAuthentication</PackageId>
-        <Description>Enables a service to authenticate to Azure services using the developer's Azure Active Directory/ Microsoft account during development, and authenticate as itself (using OAuth 2.0 Client Credentials flow) when deployed to Azure.</Description>
-        <Version>1.6.1</Version>
+        <Description>There is a newer version of this library available here: https://www.nuget.org/packages/Azure.Identity/
+          Migration guide: https://docs.microsoft.com/en-us/dotnet/api/overview/azure/app-auth-migration
```
Comment (jsquire) [inline sdk/mgmtcommon/AppAuthentication/Azure.Services.AppAuthentication/Microsoft.Azure.Services.AppAuthentication.csproj:5]:
```suggestion
          Migration guide: https://docs.microsoft.com/dotnet/api/overview/azure/app-auth-migration
```

File: sdk/mgmtcommon/AppAuthentication/Azure.Services.AppAuthentication/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -2,10 +2,13 @@
 using System.Runtime.CompilerServices;
 
 [assembly: AssemblyTitle("Microsoft.Azure.Services.AppAuthentication")]
-[assembly: AssemblyDescription("Enables a service to authenticate to Azure services using the developer's Azure Active Directory/ Microsoft account during development, and authenticate as itself (using OAuth 2.0 Client Credentials flow) when deployed to Azure.")]
+[assembly: AssemblyDescription("There is a newer version of this library available here: https://www.nuget.org/packages/Azure.Identity/ " +
+    "Migration guide: https://docs.microsoft.com/en-us/dotnet/api/overview/azure/app-auth-migration " +
```
Comment (jsquire) [inline sdk/mgmtcommon/AppAuthentication/Azure.Services.AppAuthentication/Properties/AssemblyInfo.cs:6]:
```suggestion
    "Migration guide: https://docs.microsoft.com/dotnet/api/overview/azure/app-auth-migration " +
```

File: sdk/mgmtcommon/AppAuthentication/README.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 ## Microsoft.Azure.Services.AppAuthentication Library
 
+### Note
+This library is no longer recommended to use with the new Azure SDK. It is replaced with the new **Azure Identity client library** available for .NET, Java, TypeScript and Python and should be used for all new development. Information about how to migrate to `Azure.Identity` can be found here: [AppAuthentication to Azure.Identity Migration Guidance](https://docs.microsoft.com/en-us/dotnet/api/overview/azure/app-auth-migration).
```
Comment (jsquire) [inline sdk/mgmtcommon/AppAuthentication/README.md:4]:
```suggestion
This library is no longer recommended to use with the new Azure SDK. It is replaced with the new **Azure Identity client library** available for .NET, Java, TypeScript and Python and should be used for all new development. Information about how to migrate to `Azure.Identity` can be found here: [AppAuthentication to Azure.Identity Migration Guidance](https://docs.microsoft.com/dotnet/api/overview/azure/app-auth-migration).
```

File: sdk/mgmtcommon/AppAuthentication/Azure.Services.AppAuthentication/TokenProviders/ClientCertificateAccessTokenProvider.cs [hunk 4]
```diff
@@ -194,8 +194,8 @@ await _authenticationContext.AcquireTokenAsync(authority, resource, certCred)
                             {
                                 PrincipalUsed.CertificateThumbprint = cert.Thumbprint;
                                 PrincipalUsed.IsAuthenticated = true;
-                                PrincipalUsed.TenantId = AccessToken.Parse(accessToken).TenantId;
-
+                                PrincipalUsed.TenantId = AccessToken.TenantIdInfo(accessToken);
+
```
Comment (jsquire) [inline sdk/mgmtcommon/AppAuthentication/Azure.Services.AppAuthentication/TokenProviders/ClientCertificateAccessTokenProvider.cs:198]:
```suggestion

```

File: sdk/mgmtcommon/AppAuthentication/Azure.Services.AppAuthentication/AccessToken.cs [hunk 5]
```diff
@@ -99,6 +99,24 @@ public override string ToString()
             return _accessToken;
         }
 
+        /// <summary>
+        /// Return the deserialized tenantId if accessible
+        /// </summary>
+        /// /// <param name="accessToken">AccessToken to deserialize.</param>
```
Comment (jsquire) [inline sdk/mgmtcommon/AppAuthentication/Azure.Services.AppAuthentication/AccessToken.cs:105]:
```suggestion
        /// <param name="accessToken">AccessToken to deserialize.</param>
```

File: sdk/mgmtcommon/AppAuthentication/Azure.Services.AppAuthentication/AccessToken.cs [hunk 6]
```diff
@@ -99,6 +99,24 @@ public override string ToString()
...
+        /// <returns></returns>
```
Comment (jsquire) [inline sdk/mgmtcommon/AppAuthentication/Azure.Services.AppAuthentication/AccessToken.cs:106]: Should we consider filling this out or removing it?

**Review by jsquire — CHANGES_REQUESTED** [review]

**Comment by jsquire** [comment]
@weshaggard:  There is a permissions error with the tag step - and it seems odd to me that we'd be tagging for a CI run associated with a PR.   I'm not overly familiar with the management pipeline.   I'd appreciate your thoughts.

**Comment by weshaggard** [comment]
I don't believe that the "Azure Stack MultiAPI Only" pipeline should be running on this PR. @ArcturusZhang it looks like you setup this pipeline can you please confirm whether or not it should be running?

File: sdk/mgmtcommon/AppAuthentication/Azure.Services.AppAuthentication/AccessToken.cs (hunk 6 above)
Comment (anaismiller) [reply sdk/mgmtcommon/AppAuthentication/Azure.Services.AppAuthentication/AccessToken.cs:106]: I copied the format used in other files in this project ie: https://github.com/Azure/azure-sdk-for-net/blob/6052a04aa67333828d3c882fe550ee1f2fc9ed40/sdk/mgmtcommon/AppAuthentication/Azure.Services.AppAuthentication/TokenProviders/ClientCertificateAccessTokenProvider.cs#L121

Comment (jsquire) [reply sdk/mgmtcommon/AppAuthentication/Azure.Services.AppAuthentication/AccessToken.cs:106]: The type is internal, so not a big concern.

**Review by jsquire — APPROVED** [review]
Code changes look good to me.    We'll want to wait on the pipeline conversation to resolve before we look at merging.

**Pull request merged**